### PR TITLE
OGM-716 FIx distirbution package

### PR DESCRIPTION
**Pull request by DavideD**
Hello can you guys have alook a the resulting distribution package and let me know?

It should be similar to the previous one, except for:
- There is on optional folder containing parboiled and hibernate search dependencies
- core contains hibernate-ogm-core and all the required dependencies

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -103,11 +105,12 @@
                 <groupId>org.hibernate</groupId>
                 <artifactId>hibernate-envers</artifactId>
                 <version>${hibernateVersion}</version>
-            </dependency>
```
Comment (gunnarmorling) [inline bom/pom.xml]: Why did you remove this module from the BOM?

File: distribution/pom.xml [hunk 2]
```diff
@@ -37,10 +37,97 @@
...
+            <groupId>org.hibernate.ogm</groupId>
+            <artifactId>hibernate-ogm-javadoc</artifactId>
+            <version>${project.version}</version>
+            <classifier>javadoc</classifier>
+        </dependency>
+        <dependency>
+            <groupId>org.hibernate.ogm</groupId>
+            <artifactId>hibernate-ogm-couchdb</artifactId>
+        </dependency>
+        <dependency>
```
Comment (gunnarmorling) [inline distribution/pom.xml]: Why is Envers needed?

File: distribution/pom.xml [hunk 3]
```diff
@@ -37,10 +37,97 @@
...
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-envers</artifactId>
+        </dependency>
+
+        <dependency>
```
Comment (gunnarmorling) [inline distribution/pom.xml]: Out of interest, why are some of OGM's dependency listed here explicitly, but not all?

File: distribution/src/main/assembly/dist.xml [hunk 4]
```diff
@@ -20,254 +20,139 @@
...
-            </binaries>
-        </moduleSet>
+        <!-- Provided libraries -->
+        <dependencySet>
+            <useProjectArtifact>false</useProjectArtifact>
+            <outputDirectory>dist/lib/provided</outputDirectory>
+            <useTransitiveFiltering>false</useTransitiveFiltering>
+            <unpack>false</unpack>
+            <includes>
+                 <include>org.jboss.logging:jboss-logging-annotations</include>
```
Comment (gunnarmorling) [inline distribution/src/main/assembly/dist.xml]: We don't need the annotations. They are only needed at OGM's build time (we somewhat ab-use "provided" to emulate a "compile-only" scope for this).

File: distribution/src/main/assembly/dist.xml [hunk 5]
```diff
@@ -20,254 +20,139 @@
...
+                 <include>org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec</include>
```
Comment (gunnarmorling) [inline distribution/src/main/assembly/dist.xml]: As we expect it provided by the environment, should we bundle provided stuff at all? Was it part of the dist before?

File: modules/wildfly/pom.xml [hunk 6]
```diff
@@ -61,11 +61,6 @@
             <optional>true</optional>
         </dependency>
         <dependency>
-            <groupId>org.hibernate</groupId>
```
Comment (gunnarmorling) [inline modules/wildfly/pom.xml]: We don't need that anymore? Isn't it needed as part of putting the latest ORM version in place?

**Comment by gunnarmorling** [comment]
@DavideD, Looks very good overall, added some detail remarks. Seems like it's even cleaner than before as all the transitive deps don't need to be listed anymore.

**Comment by gunnarmorling** [comment]
The JPA API is under dist/lib/optional. Shouldn't it be under core?

File: bom/pom.xml (hunk 1 above)
Comment (DavideD) [inline bom/pom.xml]: I removed it because we don't use it anywhere? Is there any particular reason to keep it?

Anyway, I kept the change in a separate commit because I wasn't sure about it

File: distribution/src/main/assembly/dist.xml (hunk 4 above)
Comment (DavideD) [inline distribution/src/main/assembly/dist.xml]: It was part of the dist before so I kept it, I'm fine removing it

File: distribution/src/main/assembly/dist.xml (hunk 5 above)
Comment (DavideD) [inline distribution/src/main/assembly/dist.xml]: Yes, it was part of the dist before, that's why I kept it

File: modules/wildfly/pom.xml (hunk 6 above)
Comment (DavideD) [inline modules/wildfly/pom.xml]: I dont know, none of our projects requires hibernate-infinispan and we never include it in modules or dist.

File: distribution/pom.xml (hunk 2 above)
Comment (DavideD) [inline distribution/pom.xml]: I don't think it's needed I'll remove it

File: bom/pom.xml (hunk 1 above)
Comment (gunnarmorling) [inline bom/pom.xml]: The idea of the BOM is to give a user matching versions of all relevant
libs. Say you have an app using ORM@RDBMS + OGM and want to use Envers with
the former. Without that entry you'd have to match the Envers version with
the other ORM components defined by the BOM.

File: distribution/src/main/assembly/dist.xml (hunk 5 above)
Comment (gunnarmorling) [inline distribution/src/main/assembly/dist.xml]: Ok, cool.

File: modules/wildfly/pom.xml (hunk 6 above)
Comment (gunnarmorling) [inline modules/wildfly/pom.xml]: +1 for not adding it to the dist.

But in case of the app server ZIPs, omitting it would leave behind the user
with a "half" ORM version, potentially falling onto their feet in case
they'd like to use ISPN as L2 cache. Not sure whether anyone is doing it,
but I'd leave this as is.

File: bom/pom.xml (hunk 1 above)
Comment (DavideD) [inline bom/pom.xml]: Ok, I'll keep it in the BOM

File: distribution/pom.xml (hunk 3 above)
Comment (DavideD) [inline distribution/pom.xml]: It's related to the way the assembly plugin works when `useTransitiveFiltering` is set to true.

Those are the dependencies that are optional or shared among different backends,
I need them as direct dependencies of the project to include them in separates folder using the dependencySet in the assembly.

File: modules/wildfly/pom.xml (hunk 6 above)
Comment (DavideD) [inline modules/wildfly/pom.xml]: This is only the declaration of the dependency, we are not using it in the assembly that creates the module for WIldFly.
WildFly already contains hibernate-infinispan in the ORM module.

The only issue I can see is that WildFly 8.2 contains ORM 4.3.7 instead of 4.3.8 but these does not seems to cause problems at the moment.

This does not seem to cause issues, I would wait until they upgrade the dependencies in WIldFly

File: distribution/pom.xml (hunk 3 above)
Comment (gunnarmorling) [inline distribution/pom.xml]: Ah, I see. Probably worth a short comment.

**Comment by DavideD** [comment]
@gunnarmorling 
Hi, I've applied your remarks and squashed the commit:
- I've added back hibernate-infinispan in the pom
- I've moved jboss-logging and hibernate-jpa-2.1-api in core
